src/AWSS3.jl

```diff
@@ -195,19 +195,38 @@ end
 
 s3_get_meta(a...; b...) = s3_get_meta(global_aws_config(), a...; b...)
 
+function _s3_exists_file(aws::AbstractAWSConfig, bucket, path; kw...)
+    q = Dict("prefix" => path, "delimiter" => "", "max-keys" => 1)
+    l = S3.list_objects_v2(bucket, q; aws_config=aws)
+    c = get(l, "Contents", nothing)
+    c ≡ nothing && return false
+    return get(c, "Key", "") == path
+end
+
+function _s3_exists_dir(aws::AbstractAWSConfig, bucket, path; kw...)
+    a = chop(string(path)) * "."
```

Comment:
Can you transfer some of your comments from the PR to code comments here, to explain what’s going on with the `.`?